### DISCUSSION TITLE
Raspberry Pi installation

### DISCUSSION THREAD

**Discussion by Binsoma** [coreemu/core#535] [Q&A]
Hi,

I need to undrstand whats a problem here, I am getting the following error while installing on raspberry Pi.

pi@raspberrypi:~/core$ sudo ./install.sh
setup to install CORE using apt
Reading package lists... Done
Building dependency tree
Reading state information... Done
python3-venv is already the newest version (3.7.3-1).
python3-pip is already the newest version (18.1-5+rpt1).
The following packages were automatically installed and are no longer required:
  libexiv2-14 libgfortran3 libgmime-2.6-0 libncurses5 libssl1.0.2 uuid-dev
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: pipx in /root/.local/lib/python3.9/site-packages (0.15.6.0)
Requirement already satisfied: userpath>=1.4.1 in /root/.local/lib/python3.9/site-packages (from pipx) (1.4.1)
Requirement already satisfied: packaging>=20.0 in /root/.local/lib/python3.9/site-packages (from pipx) (20.7)
Requirement already satisfied: argcomplete<2.0,>=1.9.4 in /root/.local/lib/python3.9/site-packages (from pipx) (1.12.2)
Requirement already satisfied: distro; platform_system == "Linux" in /root/.local/lib/python3.9/site-packages (from userpath>=1.4.1->pipx) (1.5.0)
Requirement already satisfied: click in /root/.local/lib/python3.9/site-packages (from userpath>=1.4.1->pipx) (7.1.2)
Requirement already satisfied: pyparsing>=2.0.2 in /root/.local/lib/python3.9/site-packages (from packaging>=20.0->pipx) (2.4.7)
WARNING: You are using pip version 20.2.3; however, version 20.3.1 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
/root/.local/bin has been been added to PATH, but you need to open a
    new terminal or re-login for this PATH change to take effect.
/root/.local/bin has been been added to PATH, but you need to open a
    new terminal or re-login for this PATH change to take effect.

You will need to open a new terminal or re-login for the PATH changes
to take effect.

Otherwise pipx is ready to go! ✨ 🌟 ✨
'invoke' already seems to be installed. Not modifying existing installation in '/root/.local/pipx/venvs/invoke'. Pass '--force' to force installation.
installing core locally: False
installing core with prefix: /usr/local
checking for old installations ... done
installing system dependencies ... done
installing system grpcio-tools ... done
building core ... done
installing vcmd/gui ... done
installing core virtual environment ... |Encountered a bad command exit code!

Command: 'cd daemon && poetry install --no-dev'

Exit code: 1

Stdout:

      1070│                 output = subprocess.check_output(
      1071│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1072│                 )
      1073│         except CalledProcessError as e:
    → 1074│             raise EnvCommandError(e, input=input_)
      1075│
      1076│         return decode(output)
      1077│
      1078│     def execute(self, bin, *args, **kwargs):

Stderr:

                                                                               pi@raspberrypi:~/core$

**Comment by bharnden** [comment]
Looks like an issue related to installing the poetry environment/dependencies. No one I know of has installed this on a raspberry pi, so it potentially is very likely due to that alone.

You may be able to rerun the failed poetry install command above with more verbose output to see where it is failing, but it may lead to a dead end.
